### PR TITLE
Prevent Broken ledger live for celo users

### PR DESCRIPTION
**Pull request by aaronmgdr**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.**  => dependency update no function change
- [ ] **Impact of the changes:** 
     - celo specific features. such as voting for validators, and staking

### 📝 Description

It is necessary to upgrade celo dependencies for ledger live to continue working with Celo once the Celo transition to Layer 2 happens. This is because early versions of celo blockchain used a custom transaction type and older versions of celo sdks only build transictions using this deprecated transaction format. However cel2 will not support this transaction format. 

### ❓ Context

closes https://github.com/LedgerHQ/ledger-live/issues/7551

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .pnpmfile.cjs [hunk 1]
```diff
@@ -62,20 +62,6 @@ function readPackage(pkg, context) {
       addPeerDependencies("@storybook/addon-react-native-web", {
         webpack: "*",
       }),
-      /* @celo/* packages */
-      addDependencies(/@celo\/(?!base)+/, { "@celo/base": `^${pkg.version}` }),
```
Comment (aaronmgdr) [inline .pnpmfile.cjs:66]: this is no longer necessary

**Comment by aaronmgdr** [comment]
perplexed as to why in the ledger live environment calling toChecksumAddress results in eventually the @noble/hash library asserting that the Buffer of the address is a Uint8Array and (this is the crucial part) it gives false. whereas i can trace the same code path in a different environment and it will return true. I made an empty electron repo with just ContractKit as I thought it could be the differenc but received no issue.

**Comment by aaronmgdr** [comment]
This fails because in JSDom Buffer does not inherit from Uint8Array. this causes a cascade of failures starting in @noble/hashes and bubbling up to  web3.js `new Contract(`) which is called from `newKit` (from contractkit)

Its a bit of a bummer. And im not sure of the best approach to fix it. 

As an alternative I [played around with removing contractkit completely](https://github.com/aaronmgdr/ledger-live/pull/1). But that too requires a bit more to get it working, and will change a substantial amount of files. So i wont do more on that just  yet.

**Comment by aaronmgdr** [comment]
@Wozacosta id love to talk more about how we can get this working.

**Comment by aaronmgdr** [comment]
do not close this pr

**Comment by qperrot** [comment]
Hello @aaronmgdr,
I am trying your PR* locally, and have issues when trying to do any actions using my Celo accounts (sending, withdrawing rewards).
Did you face the same issue on your side ?
Here a screen shot of the error:
Screenshot 2024-09-27 at 10 16 53

Also, the @celo packages that are being updated have this error message: Versions less than 5.1 are deprecated and will no longer be able to submit transactions to celo in a future hardfork
Do you have a timeline or a date that we should have in mind to judge of the urgency.

**Comment by aaronmgdr** [comment]
> Hello @aaronmgdr, I am trying your PR* locally, and have issues when trying to do any actions using my Celo accounts (sending, withdrawing rewards). Did you face the same issue on your side ? Here a screen shot of the error: Screenshot 2024-09-27 at 10 16 53
> 
> Also, the @celo packages that are being updated have this error message: Versions less than 5.1 are deprecated and will no longer be able to submit transactions to celo in a future hardfork Do you have a timeline or a date that we should have in mind to judge of the urgency.

@qperrot thanks for getting back. to be honest i had trouble understanding how to get the ledger system running. Hoping we can work together on this. It was tricky upgrade so far. 

As for the timeline mentioned. It is the same as the celo transition to L2. The Testnet was just [upgraded today ](https://x.com/Celo/status/1839374149095190677) and the plan for [celo mainnet is by end of year](https://docs.celo.org/cel2)

File: apps/ledger-live-desktop/jest.config.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ const commonConfig = {
     __SENTRY_URL__: null,
     __PRERELEASE__: "null",
     __CHANNEL__: "null",
+    Buffer: Uint8Array,
```
Comment (aaronmgdr) [inline apps/ledger-live-desktop/jest.config.js:40]: this was necessary to fix issue where jsdom doesn't really support Buffer.

@noble/curve relies on Buffer being a subclass of UINT8Array. the implementation in JSDOM does not.

File: .pnpmfile.cjs [hunk 3]
```diff
@@ -63,19 +63,9 @@ function readPackage(pkg, context) {
...
+        "web3": pkg.peerDependencies?.web3 ?? "1.10",
       }),
-      addDependencies("@celo/contractkit", {
-        "web3-utils": pkg.dependencies?.["web3"],
-      }),
-      addDependencies("@celo/utils", {
-        "fp-ts": "*",
-        rlp: "*",
-      }),
-      /*  @cosmjs/* packages */
```
Comment (aaronmgdr) [inline .pnpmfile.cjs:78]: todo add back

**Comment by aaronmgdr** [comment]
DO NOT MERGE WITHOUT DOING A DEPLOYMENT TO STAGING!

**Comment by aaronmgdr** [comment]
Tested with Ledger using celo-spender app 1.1.10 

- sending ✅
- locking ✅
- voting ✅

**Comment by nicolasbrugneaux** [comment]
Tested with Ledger using locally built celo-spender app 1.2.3

sending ✅
locking ✅
voting ✅

**Comment by vforgeoux-ledger** [comment]
Can it be tested with Celo app 1.3.1 ?

**Comment by nicolasbrugneaux** [comment]
Yes!

File: apps/ledger-live-desktop/jest.config.js (hunk 2 above)
Comment (nicolasbrugneaux) [reply apps/ledger-live-desktop/jest.config.js]: resolved as of [10be86b](https://github.com/LedgerHQ/ledger-live/pull/7622/commits/10be86b6acf5ce8629a41aa6a74c5ec832e78883)

File: .pnpmfile.cjs (hunk 3 above)
Comment (aaronmgdr) [inline .pnpmfile.cjs:78]:
```suggestion
      /*  @cosmjs/* packages */
```

File: .pnpmfile.cjs [hunk 4]
```diff
@@ -63,19 +63,9 @@ function readPackage(pkg, context) {
...
       addDependencies("@cosmjs/proto-signing", {
```
Comment (aaronmgdr) [inline .pnpmfile.cjs:69]:
```suggestion
      /*  @cosmjs/* packages */
      addDependencies("@cosmjs/proto-signing", {
```

File: libs/ledger-live-common/src/families/celo/hw-app-celo.ts [hunk 5]
```diff
@@ -1,20 +1,92 @@
-import type Transport from "@ledgerhq/hw-transport";
+/* eslint-disable no-console */
```
Comment (aaronmgdr) [inline libs/ledger-live-common/src/families/celo/hw-app-celo.ts:1]:
```suggestion
```

File: libs/ledger-live-common/src/families/celo/hw-app-celo.ts [hunk 6]
```diff
@@ -50,7 +123,7 @@ export class CeloApp {
       } else {
         rawTx.copy(buffer, 0, offset, offset + chunkSize);
       }
-
+      console.info("buffer buffer buffer", buffer.toString());
```
Comment (aaronmgdr) [inline libs/ledger-live-common/src/families/celo/hw-app-celo.ts:129]:
```suggestion

File: libs/ledger-live-common/src/families/celo/signOperation.ts [hunk 7]
```diff
@@ -1,14 +1,15 @@
+/* eslint-disable no-console */
```
Comment (aaronmgdr) [inline libs/ledger-live-common/src/families/celo/signOperation.ts:1]:
```suggestion
```

File: libs/ledger-live-common/src/families/celo/signOperation.ts [hunk 8]
```diff
@@ -28,31 +29,42 @@ export const signOperation: AccountBridge<Transaction, CeloAccount>["signOperati
...
+            celo.verifyTokenInfo(to!, chainId!),
+            celo.determinePrice(unsignedTransaction),
+          ]);
 
-          o.next({ type: "device-signature-requested" });
+          const rlpEncodedTransaction = await celo.rlpEncodedTxForLedger(unsignedTransaction);
 
+          o.next({ type: "device-signature-requested" });
+          // celo.signTransaction already does the eip155 v chain stuff
+          // TODO should we be using clearSignTransaction like evm fam does?
```
Comment (aaronmgdr) [inline libs/ledger-live-common/src/families/celo/signOperation.ts:45]:
```suggestion
```

File: libs/ledger-live-common/src/families/celo/signOperation.ts [hunk 9]
```diff
@@ -81,13 +93,18 @@ export const signOperation: AccountBridge<Transaction, CeloAccount>["signOperati
 
 const trimLeading0x = (input: string) => (input.startsWith("0x") ? input.slice(2) : input);
 
+// this is where it all goes wrong
+// we need to to the conversion from string to number / Buffer. the v seems clearly wrong.
+// Buffers im less sure of. might need to do 0x prefix before converting
+// v keeps on parsing as 8.123437e1341 or some bs. which seems wrong. it should be an INT. not a decimal.
```
Comment (aaronmgdr) [inline libs/ledger-live-common/src/families/celo/signOperation.ts:99]:
```suggestion
```

**Comment by aaronmgdr** [comment]
closing in favor of #8240

**Pull request closed without merging**